### ISSUE TITLE
bug(mat-sidenav): There is no focus trap when Sidenav position is 'end'

### ISSUE THREAD

**Issue by avishome** [angular/components#26572]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

11

### Description

When you opens the Sidenav in over mode the focus shouldn't leave the Sidenav when it is open. it works in case the Sidenav opens in the left side. but in the right side it is broken.

### Reproduction

Steps to reproduce:
[stackBlitz](https://stackblitz.com/edit/angular-96vr4f?file=src%2Fapp%2Fsidenav-mode-example.html)
1. Open the Stackblitz example
2. choose Mode: Over
3. click on the toggle button
4. In the keyboard, click on the tab several times.

### Expected Behavior

The focus shouldn't leave the Sidenav.

### Actual Behavior

The focus leaves the side pane and there is no way to return to Sidenav with the keyboard.

### Environment

- Angular: 14.2.10
- CDK/Material: 14.2.7
- Browser(s): Chrome
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by zarend** [comment]
Confirm that I this happens with v15. That's because the stackblitz uses v15
